dioptas/widgets/plot_widgets/utils.py

```diff
@@ -19,27 +19,146 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 from __future__ import annotations
 
+import re
+from typing import Optional
 import numpy as np
 
 
-def auto_level(hist_x: np.ndarray, hist_y: np.ndarray) -> tuple[float, float]:
-    """Compute colormap range from histogram of the data
+def _default_auto_level(data: np.ndarray) -> tuple[float, float]:
```

Comment:
Using an implementation of the default autoscale that first computes the histogram.
This spares passing both data and histogram around at the expense of re-computing the histogram